### PR TITLE
Add `Constructor` annotation

### LINKED ISSUE #10617: suggestion for Crystal docs: File.open(... &) to be in constructor section
In the [API for File](https://crystal-lang.org/api/1.0.0/File.html), the [constructor](https://crystal-lang.org/api/1.0.0/File.html#constructors) lists: `File.new`, `File.open`

Meanwhile, the overloaded version of `File.open(... &)` that accepts a block is excluded from the constructor section and 2/3s of the way down the page.  I feel that overloads should stick together as much as possible, especially when the examples in the [Overview section](https://crystal-lang.org/api/1.0.0/File.html#overview) at the top of the page includes the use of `File.open(... &)` 

Specifically, I think it would be more beginner-friendly if `File.open(... &)` were included in the constructor section.

### PR DESCRIPTION
**Pull request by Blacksmoke16**
Resolves #10617

### CODE REVIEW

File: src/string/utf16.cr [hunk 1]
```diff
@@ -106,6 +106,7 @@ class String
   #
   # Invalid values are encoded using the unicode replacement char with
   # codepoint `0xfffd`.
+  @[Constructor]
```
Comment (Blacksmoke16) [inline src/string/utf16.cr:109]: Um, is this a case where it's trying to compile the program using an older Crystal version that doesn't have the annotation defined? Guess the solution is to do a version check?

Comment (Blacksmoke16) [reply src/string/utf16.cr:109]: Or possibly add the annotation, but not use it until a version with the annotation is released?

Comment (straight-shoota) [reply src/string/utf16.cr]: Huh, but `annotation Constructor` *is* defined in stdlib. So it should compile, as does this program:
```cr
annotation Constructor
end

@[Constructor]
def foo
end

foo
```

Maybe the require order is at odds?

**Comment by straight-shoota** [comment]
I have some concerns about this approach, explained in https://github.com/crystal-lang/crystal/issues/10617#issuecomment-1864423850

File: src/string/utf16.cr (hunk 1 above)
Comment (Blacksmoke16) [reply src/string/utf16.cr]: Bingo. String is required before the annotations.

**Pull request closed without merging**